tests/test_communication_manager.py

```diff
@@ -0,0 +1,163 @@
+import tests.context
+
+import pytest
+import time
+
+from core.communication_manager import CommunicationManager
+from core.utils.event_types     import RawEventTypes, ActionableEventTypes
+from core.runner                import DMLRunner
+from core.scheduler             import DMLScheduler
+from custom.keras               import get_optimizer
+from models.keras_perceptron    import KerasPerceptron
+from core.utils.dmljob          import DMLJob, serialize_job, deserialize_job
+from core.configuration         import ConfigurationManager
+
+def make_model_json():
+    m = KerasPerceptron(is_training=True)
+    model_architecture = m.model.to_json()
+    model_optimizer = get_optimizer(m.model)
+    model_json = {
+        "architecture": model_architecture,
+        "optimizer": model_optimizer
+    }
+    return model_json
+
+def make_initialize_job(model_json):
+    initialize_job = DMLJob(
+        "initialize",
+        model_json,
+        "keras",
+        hyperparams=make_hyperparams(),
+        label_column_name='label'
+    )
+    return initialize_job
+
+def make_hyperparams():
+    hyperparams = {
+        'averaging_type': 'data_size',
+        'batch_size': 4,
+        'epochs': 1,
+        'split': 0.004,
+    }
+    return hyperparams
+
+config_manager = ConfigurationManager()
```

Comment:
Can we move this block to the top before the method definitions?

Reply:
done